src/sentry/static/sentry/app/views/projects/projectContext.jsx

```diff
@@ -111,6 +112,11 @@ const ProjectContext = createReactClass({
     }
   },
 
+  componentWillUnmount() {
+    // Save last route so that we can jump back to view from settings
+    setLastRoute(this.props.location.pathname);
```

Comment:
If it's a project settings page, perhaps we should skip this

Reply:
Ah, good catch, I think the easiest way will probably be to introduce a ProjectRoot that wraps our project pages.